### ISSUE TITLE
Hidecallgraph (and similar) not exported to groups

### ISSUE THREAD

**Issue by TommyPec** [doxygen/doxygen#10644]
**Describe the bug**
This is easier than the other issues I report (I hope).
When you have a group documenting multiple functions, `\hidecallgraph` and other graph-related directive are only applied to the first function (tested with `\hidecallgraph` and `\hidecallergraph`)

**Expected behavior**
All the functions in a group should have the same result.

**Screenshots**
<img width="481" alt="Screenshot 2024-02-10 at 16 47 53" src="https://github.com/doxygen/doxygen/assets/169026/7083eaf5-61ae-4008-8ac3-fea970cb15b3">

**To Reproduce**
The above can be easily reproduced with this snippet:
```cpp
/**
 * \ingroup timecivil
 * Construct a Time in the indicated unit.
 * \param [in] value The value
 * \return The Time
 *
 * Call graph was not generated because of its size.
 * \hidecallergraph
 * \hidecallgraph
 * @{
 */
inline Time
Years(double value)
{
    return Time::FromDouble(value, Time::Y);
}

inline Time
Years(int64x64_t value)
{
    return Time::From(value, Time::Y);
}
/**@}*/ // Construct a Time in the indicated unit.
```

**Version**
Doxygen 1.11.0 (4ee0a7be0479476cb7eea214abe692bcd012d9b3*)
MacOS ARM

**Stack trace**
N/A

**Additional context**
N/A

**Comment by albert-github** [comment]
> This is easier than the other issues I report (I hope).

Which other issues do you mean? Did you mean #10636 ?
- when this is the case please add this issue #10644 to #10636 and also retest with 
  ```
  Commit: 0588c3220faef842ac8c3ebf63bc0fe4fe2a800c [0588c32]
  Date: Friday, February 9, 2024 6:05:35 PM
  
  issue #10636 weird behavior with `\hideinheritancegraph`
  ``` 
  as in the latest PR update of the issue #10636 / #10638 a.o. the `\hidecallgraph` and `\hidecallergraph` are handled.

With the example I cannot reproduce the problem as:
- I get a warning:
  ```
  nstime.h:23: warning: Member Years(int64x64_t value) (function) of group timecivil is not documented.
  ```
  (and I already added `\defgroup timecivil ...`)
- The function `Years` is not called / doesn't call anything known.

so when the retest with 0588c3220faef842ac8c3ebf63bc0fe4fe2a800c does not give a correct result:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by albert-github** [comment]
#10638 has been integrated into master on GitHub, best to test against the master version.

**Comment by TommyPec** [comment]
> > This is easier than the other issues I report (I hope).
> 
> Which other issues do you mean? Did you mean #10636 ?

Nope, I was referring to #10626, which I consider weird (and have nothing to do with this one).

> 
> With the example I cannot reproduce the problem as:
> 
> * I get a warning:
>   ```
>   nstime.h:23: warning: Member Years(int64x64_t value) (function) of group timecivil is not documented.
>   ```

Strange, it's documented... but I'm working to squash doxygen errors on a branch, so I might not have pushed some changes. I'll have to split some of them and push.

Anyway, here's a small example for this one:
[DoxyTestGraphsGrouped.tgz](https://github.com/doxygen/doxygen/files/14233080/DoxyTestGraphsGrouped.tgz)

Check "Topics/standard bool units". The function `Years` doesn't have the graphs (correct), but all the other functions do have them (incorrect). `Years` is the first function of the group. The other functions have the correct documentation (i.e., they're all sharing the same text), but the graphs are back.

**Comment by albert-github** [comment]
I think I now understand the problem.
The documentation is distributed over the different functions (`Years`, `Seconds`, `PicoSecond`, `FemptoSeconds`, ...) but the settings like `\hidecallergraph` is only applied with the first function (here `Years`).
It looks like these settings also have to be distributed to the other functions (but of course only when that function doesn't have an explicit "hide" / "show" command).

**Comment by TommyPec** [comment]
> I think I now understand the problem. The documentation is distributed over the different functions (`Years`, `Seconds`, `PicoSecond`, `FemptoSeconds`, ...) but the settings like `\hidecallergraph` is only applied with the first function (here `Years`). It looks like these settings also have to be distributed to the other functions (but of course only when that function doesn't have an explicit "hide" / "show" command).

Correct, that's exactly the problem.

**Comment by albert-github** [comment]
@doxygen 

It looks like a `mergeMemberOverrideOptions` in the `distributeMemberGroupDocumentation` of `membergroup.cpp` would do for the presented case, though when we place before e.g. the `Seconds()` function:
```
/**
 * \callergraph
 * \callgraph
 */
```
it looks to me that in that case for the `Seconds()` function,  in my opinion, the explicit overruling of the group setting won't be taken into account due to the fact that we cannot differentiate anymore in the test in `mergeMemberOverrideOptions` between an explicit setting and an implicit setting based on the configuration and thus the `\hidecallgraph` of the group will be taken...

**Comment by TommyPec** [comment]
> @doxygen
> 
> It looks like a `mergeMemberOverrideOptions` in the `distributeMemberGroupDocumentation` of `membergroup.cpp` would do for the presented case, though when we place before e.g. the `Seconds()` function:
> 
> ```
> /**
>  * \callergraph
>  * \callgraph
>  */
> ```
> 
> it looks to me that in that case for the `Seconds()` function, in my opinion, the explicit overruling of the group setting won't be taken into account due to the fact that we cannot differentiate anymore in the test in `mergeMemberOverrideOptions` between an explicit setting and an implicit setting based on the configuration and thus the `\hidecallgraph` of the group will be taken...

Sometimes knowledge is power, and RTFM is the answer.

I.e., the important part (in my opinion) is to clarify what is the expected behaviour and how to cope with common issues.
It would be acceptable (to me, at least) to say "commands to modify the default behaviour on graph generations are NOT allowed on group-generated documentation, you MUST do this":

However, I'm experimenting with some variations and there are drawbacks and glitches. E.g., this is "correct but wrong":
```cpp
/**
 * \ingroup boolcivil
 * Construct a bool in the indicated unit.
 * \param [in] value The value
 * \return The bool
 * @{
 */

/**
 * \hidecallergraph
 * \hidecallgraph
 * Call graph was not generated because of its size.
 */
inline void
Years(double value)
{
    std::cout << "Years(double)\n";
}

/**
 * \hidecallergraph
 * \hidecallgraph
 * Call graph was not generated because of its size.
 */
inline void
Days(double value)
{
    std::cout << "Days(double)\n";
}

/**@}*/
```

It is "correct" because the graphs are shown/hidden correctly, but it's "wrong" because the `Days` function documentation looses the part defined by the group:
<img width="412" alt="Screenshot 2024-02-11 at 12 39 20" src="https://github.com/doxygen/doxygen/assets/169026/9c51826d-4170-4c88-9b24-0a38e147014c">

If you **only** use
```cpp
/**
 * \hidecallergraph
 * \hidecallgraph
 */
```
i.e., the functions in the group have a "documentation" without any text and only `\` commands, the functions are documented in the right way, but you can't add any further text.

In other words, any text beside commands will replace the group's function documentation. This might as well be a design choice - I don't argue - provided that it's documented.
Perhaps a better solution would be to allow to substitute the group's documentation or to *continue* it.

Note: in the "wrong" case the behaviour is inconsistent. The first function (`Years`) have its documentation _concatenated_, the second (`Days`) is _substituted_.

**Comment by TommyPec** [comment]
What I was suggesting in the previous comment was a `\continue` Vs a `\substitute` directive in the documentation for grouped elements.

**Comment by TommyPec** [comment]
Sorry to bug... any update on this?

**Comment by albert-github** [comment]
There have been some updates in the respect to the handling of `\hide...` but that does not influence this issue.
It looks like that these "override commands" are not propagated in case of groups.
